### PR TITLE
Added Basic Authorization to RequestBuilder and improved unit tests.

### PR DESCRIPTION
**Pull request by ghost**
- Added BasicAuthorization and it's unit tests to the RequestBuilder.
- Extracted Validator private class in RequestBuilderTest.

### CODE REVIEW

File: commons-rest/src/main/java/com/mule/connectors/commons/rest/builder/RequestBuilder.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import com.mule.connectors.commons.rest.builder.handler.ResponseHandler;
 import com.mule.connectors.commons.rest.builder.strategy.*;
 import org.apache.commons.lang3.StringUtils;
+import org.glassfish.jersey.internal.util.Base64;
```
Comment (marianogoldman) [inline commons-rest/src/main/java/com/mule/connectors/commons/rest/builder/RequestBuilder.java]: Not sure about using internal classes...

Comment (marcosnc) [inline commons-rest/src/main/java/com/mule/connectors/commons/rest/builder/RequestBuilder.java]: We may use this:
https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base64.html

Comment (marcosnc) [inline commons-rest/src/main/java/com/mule/connectors/commons/rest/builder/RequestBuilder.java]: Another good idea could be this: http://stackoverflow.com/questions/14413169/which-java-library-provides-base64-encoding-decoding
Seems like Java 6 and 7 have OOB support for Base64.

Comment (martinpaoloni) [inline commons-rest/src/main/java/com/mule/connectors/commons/rest/builder/RequestBuilder.java]: Agreed. `java.util.Base64` should provide what you need.

**Comment by carlRcastro** [comment]
Gaston, 
Sorry to bother you, but - could you please remove me from this list??
I worked for Mulesoft briefly in 2013, but haven’t worked there since Jan. 2014, and am not a contributor to the community edition.
I thought I’d removed myself from all the lists to which I’d been subscribed, but somehow I keep popping up on a few.
Thanks!
And all the best in your good work on Mule...
Cheers,
Carl

**Comment by marianogoldman** [comment]
👍

**Pull request merged**